samples/bluetooth/central_hr/src/main.c

```diff
@@ -153,9 +167,12 @@ static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
 	printk("[DEVICE]: %s, AD evt type %u, AD data len %u, RSSI %i\n",
 	       dev, type, ad->len, rssi);
 
-	/* We're only interested in connectable events */
+	/* We're only interested in legacy connectable events or
+	 * possible extended advertising that are connectable.
+	 */
 	if (type == BT_GAP_ADV_TYPE_ADV_IND ||
-	    type == BT_GAP_ADV_TYPE_ADV_DIRECT_IND) {
+	    type == BT_GAP_ADV_TYPE_ADV_DIRECT_IND ||
+	    type == BT_GAP_ADV_TYPE_EXT_ADV) {
```

Comment:
In the case that it is extended, we should filter additionally on `BT_GAP_ADV_PROP_CONNECTABLE`

Reply:
The comment above doesn't match the code. 

This will still call `bt_data_parse` on non-connectable extended advertising, right?

Reply:
yes, but that should be ok as the connection request is only sent to peripheral with HR service listed in its services AD type

Reply:
> yes, but that should be ok as the connection request is only sent to peripheral with HR service listed in its services AD type

A device can advertise HR without being connectable. I don't see any reason why we wouldn't add the connectable check so that we don't have a device that may attempt to connect to non-connectable advertising